### PR TITLE
gh-37: Add Data Specification PDF exporter

### LINKED ISSUE #37: Add a PDF exporter
The Mauro Data Explorer (MDE) allows researchers to create data specifications. These data specifications consist of the fields that the researcher wants access to and queries to restrict the types of data that they wish to see. Currently the data specification is saved as a Mauro Data Model in a Mauro database. The data specification needs to be exportable to other systems such as the Secure Data Environment (SDE) and converted to PDF that can be easily read by SDE users.

The suggested starting point for PDF generation is to use DITA. There are already DITA plugins in Mauro, so these need to be investigated to work out how best to go about this.

### PR DESCRIPTION
**Pull request by NigelPalmer**
Closes #37 

To test this work you will need to:
1. Create a Data Specification in the MDE
2. Log out of the MDE and log into Mauro UI as admin
3. Go to the Data Specification you created in step 2
4. Click on the export icon
5. Choose "PDF Data Specification Exporter"
6. Export

Note that this is the first iteration of the PDF. We will come back and export more detailed information, but the information contained within will be enough to get the system up and running for demo purposes. Also note that some of the formatting in the PDF could be improved, but for now it's just about getting the info out there.

Tests have been written to prove this works.

### CODE REVIEW

File: grails-app/services/uk/ac/ox/softeng/maurodatamapper/plugins/explorer/pdf/exporter/core/PdfExportFormatterService.groovy [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        parseQuery(jsonObject)
+    }
+
+    /*
+    private static String meqlConnective(String connective) {
+        (connective.toLowerCase() != 'and' && connective.toLowerCase() != 'or') ?
+            'UNKNOWN_CONNECTIVE' : connective.toLowerCase()
+    }
+
+     */
```
Comment (abwilson23) [inline grails-app/services/uk/ac/ox/softeng/maurodatamapper/plugins/explorer/pdf/exporter/core/PdfExportFormatterService.groovy]: Unused code comment.

File: grails-app/services/uk/ac/ox/softeng/maurodatamapper/plugins/explorer/pdf/exporter/core/PdfExportFormatterService.groovy [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        }
+
+        if (lastRule && connective != '') {
+            meql += '\r\n' + closingTabs + ')'
+        }
+
+        meql
+    }
+
+     */
```
Comment (abwilson23) [inline grails-app/services/uk/ac/ox/softeng/maurodatamapper/plugins/explorer/pdf/exporter/core/PdfExportFormatterService.groovy]: Unused code comment.

**Review by abwilson23 — APPROVED** [review]
Code changes look sensible and it passes functional testing ✅.

Minor comments about unused code. Remove if you see fit.

File: grails-app/services/uk/ac/ox/softeng/maurodatamapper/plugins/explorer/pdf/exporter/core/PdfExportFormatterService.groovy (hunk 1 above)
Comment (NigelPalmer) [reply grails-app/services/uk/ac/ox/softeng/maurodatamapper/plugins/explorer/pdf/exporter/core/PdfExportFormatterService.groovy]: Commented out code deleted.

File: grails-app/services/uk/ac/ox/softeng/maurodatamapper/plugins/explorer/pdf/exporter/core/PdfExportFormatterService.groovy (hunk 2 above)
Comment (NigelPalmer) [reply grails-app/services/uk/ac/ox/softeng/maurodatamapper/plugins/explorer/pdf/exporter/core/PdfExportFormatterService.groovy]: Commented out code deleted.

**Pull request merged**